File: src/main/java/henaxel/raytracer/utils/ImageReader.java
package henaxel.raytracer.utils;

import javax.imageio.ImageIO;
import java.awt.image.BufferedImage;
import java.io.File;
import java.io.IOException;

public class ImageReader {

    public ImageReader() {}

    public static BufferedImage loadBufferedImage(String path) {
        BufferedImage loaded = null;
        try {
            File file = new File(path);
            loaded = ImageIO.read(file);
        } catch (IOException e) {
            System.out.println("henaxel.raytracer.utils.ImageReader: IOException loading image!");
            e.printStackTrace();
        }
        return loaded;
    }
}

File: src/henaxel/raytracer/utils/ImageReader.kt
package henaxel.raytracer.utils

import java.awt.image.BufferedImage
import java.io.File
import java.io.IOException
import javax.imageio.ImageIO

class ImageReader {

    companion object{
        fun loadBufferedImage(path: String): BufferedImage? {
            var loaded: BufferedImage? = null
            try {
                val file = File(path)
                loaded = ImageIO.read(file)
            } catch(e: IOException) {
                println("\"henaxel.raytracer.utils.ImageReader: IOException loading image!\"")
                e.printStackTrace()
            }
            return loaded
        }
    }

}